fvm/src/executor/mod.rs

```diff
@@ -43,6 +45,13 @@ pub trait Executor {
     fn flush(&mut self) -> anyhow::Result<Cid>;
 }
 
+pub trait ValidateExecutor: Executor {
+    type Validator: Kernel;
+
+    /// This is the entrypoint to validate a message from an abstract account.
+    fn validate_message(&mut self, msg: Message, sig: Vec<u8>) -> anyhow::Result<ValidateRet>;
+}
```

Comment:
We can just add this as a method to `Executor`, unless I'm missing something.

Reply:
Hm. Actually, I think we have to. `execute_message` will need to call `validate_message` (or some form of it) on execution. Basically:

- Preflight stuff.
- Abstract validation (burns gas).
- Execution (burns more gas).

That is, of course, unless we go with the two-stage validation thing. I'm going to see if we can get a talk with anorth to discuss that, we're going to want to make decisions here sooner rather than later.

Reply:
True, though in theory this doesn't need to be a super trait of Executor and could be used to assert that an executor is _only_ used for validate. Though I'm not to confident on its usefulness...

Reply:
It likely won't be useful. In general, implement as a monolith _first_, then break things into smaller pieces once the boundaries become clear. Otherwise, you'll end up with the wrong boundaries, spend a lot of time trying to make it work, give up eventually, then end up with _really_ awkward code with the wrong abstractions (speaking from lots of experience doing this).